application_sandbox/separate_stencil_usage/main.cpp

```diff
@@ -0,0 +1,649 @@
+// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+const auto& cube_data = cube_model::model;
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+namespace floor_model {
+#include "floor.obj.h"
+}
+const auto& floor_data = floor_model::model;
+
+uint32_t floor_vertex_shader[] =
+#include "floor.vert.spv"
+    ;
+
+uint32_t floor_fragment_shader[] =
+#include "floor.frag.spv"
+    ;
+
+namespace plane_model {
+#include "fullscreen_quad.obj.h"
+}
+const auto& plane_data = plane_model::model;
+
+uint32_t plane_vertex_shader[] =
+#include "plane.vert.spv"
+    ;
+
+uint32_t plane_fragment_shader[] =
+#include "plane.frag.spv"
+    ;
+
+const VkFormat kDepthStencilFormat = VK_FORMAT_D32_SFLOAT_S8_UINT;
```

Comment:
This format is not guaranteed to be available. (We have used it elsewhere, but going forward we should at least query which D/S formats are supported).

IIRC someone (maybe @apazylbe) made a change along these lines no long ago to be able to query the correct format from the application.

Reply:
Fixed.